batch/batch/front_end/front_end.py

```diff
@@ -330,6 +411,47 @@ async def get_jobs(request, userdata, batch_id):  # pylint: disable=unused-argum
     return web.json_response(resp)
 
 
+@routes.get('/api/v1alpha/batches/{batch_id}/jobs/resources')
+@rest_billing_project_users_only
+async def get_jobs_for_billing(request, userdata, batch_id):
+    """
+    Get jobs for batch to check the amount of resources used.
+    Takes a "last_job_id" and "limit" parameter.
```

Comment:
Probably describing the `limit` semantics more clearly: this could be the maximum number of results returned or the maximum job ID, so maybe another name would be better?

I'd also suggest an upper limit for `limit` ;), as otherwise you're not protecting against really expensive requests. Or maybe don't make it user-configurable, but just use a default maximum value (like the UI), as the client needs to call this repeatedly anyhow.

```suggestion
    Takes a "last_job_id" and "limit" parameter that can be used to implement paging.
```

Reply:
I thought about that a few times, a user can be responsible for the expensive request if they make it right? If it crashes batch then it's on the user.

 I am happy to include one, if you have a suggestion for a sensible upper limit?

Reply:
I kind of like the user configurable, because if we're finding my script is taking a long time, I can test different values without redeploying batch

Reply:
> If it crashes batch then it's on the user.

You should always assume worst case inputs from the user, even in non-malicious scenarios. Our backends need to be robust against this, particularly as Batch is shared infrastructure. I'd use something like 1-10k as the upper result limit; you really don't want to send individual responses that are larger than a MB or so.

> because if we're finding my script is taking a long time, I can test different values without redeploying batch

Isn't it kind of unlikely that the request overhead would play a major part in this? The query on the DB should be fairly quick with the filter on the `job_id` column? You could also request multiple batch IDs in parallel.

Reply:
Seems reasonable, I've limited between [1, 1E4] (10,000).